content/200-concepts/200-database-connectors/08-cockroachdb.mdx

```diff
@@ -82,3 +82,67 @@ The following table lists any other current known limitations of CockroachDB com
+| `TIMETZ` \| `TIME WITH TIME ZONE`            | `DateTime` |    ✔️     | `@db.Timetz(x)`                      |       |
+| `DECIMAL(p,s)` \| `NUMERIC(p,s)`, `DEC(p,s)` | `Decimal`  |    ✔️     | `@db.Decimal(x, y)`                  |       |
+| `REAL` \| `FLOAT4`, `FLOAT`                  | `Float`    |    ✔️     | `@db.Float4`                         |       |
+| `DOUBLE PRECISION` \| `FLOAT8`               | `Float`    |    ✔️     | `@db.Float8`                         |       |
+| `INT2` \| `SMALLINT`                         | `Int`      |    ✔️     | `@db.Int2`                           |       |
+| ``INT4`                                      | `Int`      |    ✔️     | `@db.Int4`                           |       |
+| `SERIAL2` \| `SMALLSERIAL`                   | `Int`      |    ✔️     | `@db.Int2 @default(autoincrement())` |       |
+| `SERIAL4` \| `SERIAL`                        | `Int`      |    ✔️     | `@db.Int4 @default(autoincrement())` |       |
+| `SERIAL8` \| `BIGSERIAL`                     | `Int`      |    ✔️     | `@db.Int8 @default(autoincrement()`  |       |
+| `CHAR(n)` \| `CHARACTER(n)`                  | `String`   |    ✔️     | `@db.Char(x)`                        |       |
+| `STRING` \| `TEXT`, `VARCHAR`                | `String`   |    ✔️     | `@db.String`                         |       |
+| `DATE`                                       | `DateTime` |    ✔️     | `@db.Date`                           |       |
+| `ENUM`                                       | `Enum`     |    ✔️     | N/A                                  |       |
+| `INET`                                       | `String`   |    ✔️     | `@db.Inet`                           |       |
+| `BIT(n)`                                     | `String`   |    ✔️     | `@Bit(x)`                            |       |
+| `VARBIT(n)` \| `BIT VARYING(n)`              | `String`   |    ✔️     | `@VarBit`                            |       |
+| `OID`                                        | `Int`      |    ✔️     | `@db.Oid`                            |       |
+| `UUID`                                       | `String`   |    ✔️     | `@db.Uuid`                           |       |
+| `JSONB` \| `JSON`                            | `Json`     |    ✔️     | `@db.JsonB`                          |       |
+| `jsonb`                                      | `Json`     |    ✔️     | `@db.JsonB`                          |       |
+| `xml`                                        | `String`   |    ✔️     | `@db.Xml`                            |       |
+| Array types                                  | `[]`       |    ✔️     |                                      |       |
+
+[Introspection](/concepts/components/introspection) adds native database types that are **not yet supported** as [`Unsupported`](/reference/api-reference/prisma-schema-reference#unsupported) <span class="api"></span> fields:
+
+```prisma file=schema.prisma
+model Device {
```

Comment:
Do we have a relevant example of any unsupported CockroachDB type?

Reply:
I think we don't support `INTERVAL` (full list there: https://www.cockroachlabs.com/docs/stable/data-types.html)

Reply:
The convention we have in migrate is to always capitalize, except for edge cases (`"char"` is the only one that comes to mind).